packages/web/src/shared/components/SubmitButton.tsx

```diff
@@ -0,0 +1,28 @@
+import { ComponentProps } from 'react';
+import { useFormState } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import Button from './Button';
+import LoadingSpinner from './LoadingSpinner';
+
+export type SubmitButtonProps = Omit<ComponentProps<typeof Button>, 'type'>;
+
+export default function SubmitButton({
+  children,
+  ...props
+}: SubmitButtonProps) {
+  const { isSubmitting } = useFormState();
+  const { t } = useTranslation();
+
+  return (
+    <Button {...props} type="submit">
+      {isSubmitting ? (
+        <>
+          <LoadingSpinner className="inline-block mr-2" />
+          {t('submitting')}
+        </>
+      ) : (
+        children
+      )}
```

Comment:
Finally, the submit button can use the form helpers to show a loading spinner while the form is submitting